clientv3/retry.go

```diff
@@ -54,14 +63,50 @@ func (c *Client) newRetryWrapper() retryRpcFunc {
 	}
 }
 
+func (c *Client) newRdonlyRetryWrapper() retryRpcFunc {
+	return func(rpcCtx context.Context, f rpcFunc) error {
+		for {
+			err := f(rpcCtx)
+			if err == nil {
+				return nil
+			}
+
+			// always stop retry on etcd errors other than invalid auth token
+			eErr := rpctypes.Error(err)
+			if etcdErr, ok := eErr.(rpctypes.EtcdError); ok {
```

Comment:
instead of nesting if's, seems simpler to do:
```go
if rpctypes.Error(err) ==  rpctypes.EtcdInvalidAuthToken {
    if c.getToken(rpcCtx) != nil {
        return err
    }
    continue
}
```